examples/plugins/src/file-size.plugin.unit.test.ts

```diff
@@ -0,0 +1,357 @@
+import { unlink } from 'fs/promises';
+import { vol } from 'memfs';
+import { basename, join } from 'path';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
+import { MEMFS_VOLUME } from '@code-pushup/testing-utils';
+import { formatBytes } from '@code-pushup/utils';
+import {
+  assertFileSize,
+  errorMessage,
+  fileSizeIssues,
+  infoMessage,
+  runnerFunction,
+  scoreFilesizeAudit,
+} from './file-size.plugin';
+
+// Mock file system API's
+vi.mock('fs', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs;
+});
+vi.mock('fs/promises', async () => {
+  const memfs: typeof import('memfs') = await vi.importActual('memfs');
+  return memfs.fs.promises;
+});
+
+const outputDir = MEMFS_VOLUME;
+const projectJson = JSON.stringify(
+  {
+    test: 42,
+    arr: [1, 2, 3],
+    obj: { test: 42 },
+  },
+  null,
+  2,
+);
+const testJs = `
+    const str = 'Hello World'
+    const num = 42;
+    const obj = ${projectJson};
+  `;
+const readmeMd = '# Docs';
+
+const file = 'test.js';
+
+describe('infoMessage', () => {
+  it.each([['index.js'], [join('src', 'index.js')]])(
+    'should return info message',
```

Comment:
You should reference the test arguments in the title, otherwise you generate duplicate test titles:

```suggestion
    'should return info message for file %s',
```